shared/utility/math/euler.hpp

```diff
@@ -168,6 +168,92 @@ namespace utility::math::euler {
         return quat.matrix();
     }
 
+    /**
+     * @brief Convert rotation matrix to euler angles Thetaxy = [Roll, Pitch, Yaw]
+     */
+    template <typename Scalar>
+    void rot2rpy(const Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& Rxy,
```

Comment:
You assert it's 3 by 3 anyway so why no let the compiler take care of making sure it's 3 by 3 rather than runtime? And change this to <Scalar, 3, 3>